src/components/BookmarksBar/index.tsx

```diff
@@ -24,16 +24,33 @@ const BookmarksBar: React.FC<BookmarksBarProps> = ({
   scrollToLine,
 }) => {
   const { sendEvent } = useLogWindowAnalytics();
+  const [searchParams, setSearchParams] = useQueryParams();
 
   const [selectedLine] = useQueryParam<number | undefined>(
     QueryParams.SelectedLine,
     undefined
   );
+  const [shareLine] = useQueryParam<number | undefined>(
+    QueryParams.ShareLine,
+    undefined
+  );
   const [bookmarks, setBookmarks] = useQueryParam<number[]>(
     QueryParams.Bookmarks,
     []
   );
 
+  // If selectedLine is in the URL, replace it with shareLine.
+  // This code can be deleted in EVG-18748.
+  useEffect(() => {
+    if (selectedLine) {
+      setSearchParams({
+        ...searchParams,
+        shareLine: selectedLine,
+        selectedLine: undefined,
+      });
+    }
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Can we move this to a more generic location? Somewhere like `LogView` or `LogWindow`? Several parts of the app use `selectedLine` so it seems odd to me to handle this transformation in a random component.

Reply:
I moved it to `LogContext`, which may be a little awkward but works the best because the URL param gets replaced while the log is still downloading. This allows the jump to line functionality to work again, because `shareLine` is finished being overwritten by the time the log loads. Since this is only a temporary solution that will eventually be deleted, I think it's okay to have it in `LogContext`.